src/Step/Deploy/InstallUpdate/ConfigUpdate/DbConnection.php

```diff
@@ -60,83 +64,321 @@ class DbConnection implements StepInterface
+            if (isset($connectionsConfig2[$connectionName])
+                && !$this->isSameConnection($connectionsConfig2[$connectionName], $connectionData)) {
+                $differentConnections[] = $connectionName;
+            }
+        }
+        return $differentConnections;
+    }
 
-        if (!$connectionData->getHost()
-            || !$this->stageConfig->get(DeployInterface::VAR_MYSQL_USE_SLAVE_CONNECTION)
-            || (!$this->configMerger->isEmpty($envDbConfig) && !$this->configMerger->isMergeRequired($envDbConfig))
-        ) {
-            return;
+    /**
+     * Checks connection parameters for equality
+     *
+     * @param array $connection1
+     * @param array $connection2
+     * @return bool
+     */
+    private function isSameConnection(array $connection1, array $connection2): bool
+    {
+        return $connection1[DbConfig::KEY_HOST] === $connection2[DbConfig::KEY_HOST];
+    }
+
+    /**
+     * Returns database configuration with default and slave connections
+     *
+     * @param bool $withSlave
+     * @return array
+     * @throws ConfigException
+     */
+    private function getMainDbConfig(bool $withSlave): array
+    {
+        $dbConfig = $this->getDbConfigData();
+        $dbConfig[DbConfig::KEY_CONNECTION] = $this->getMainConnections($dbConfig[DbConfig::KEY_CONNECTION]);
+
+        if ($withSlave) {
+            $slaveConnections = $dbConfig[DbConfig::KEY_SLAVE_CONNECTION];
+            $dbConfig[DbConfig::KEY_SLAVE_CONNECTION] = $this->getMainConnections($slaveConnections);
+        } else {
+            unset($dbConfig[DbConfig::KEY_SLAVE_CONNECTION]);
         }
 
-        if (!$this->mergedConfig->isDbConfigurationCompatibleWithSlaveConnection()) {
-            $this->logger->warning(
-                'You have changed db configuration that not compatible with default slave connection.'
-            );
+        return $dbConfig;
+    }
+
+    /**
+     * Returns main resource configuration
+     *
+     * @return array
+     */
+    private function getMainResourceConfig(): array
+    {
+        return array_intersect_key(
+            $this->resourceConfig->get(),
+            array_flip([ResourceConfig::RESOURCE_DEFAULT_SETUP])
+        );
+    }
+
+    /**
+     * Returns main connection configurations
+     *
+     * @param array $connections
+     * @return array
+     */
+    private function getMainConnections(array $connections): array
+    {
+        return array_intersect_key(
+            $connections,
+            array_flip(DbConfig::MAIN_CONNECTIONS)
+        );
+    }
+
+    /**
+     * Adds logging about slave connection.
+     *
+     * @param array $dbConfig
+     * @param bool $isUseSlave
+     * @throws ConfigException
+     */
+    private function addLoggingAboutSlaveConnection(array $dbConfig, bool $isUseSlave)
+    {
+        $customDbConfig = $this->stageConfig->get(DeployInterface::VAR_DATABASE_CONFIGURATION);
```

Comment:
Better to check `if (!$isUseSlave) { return; }` at the beginning of function for avoiding not necessary operations if its value is false.

Reply:
Or remove the second argument at all and call this function only when $useSlave is true.

Reply:
Done

Reply:
Done